raitools/raitools/_managers/base_manager.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the base class for managers."""
+
+from abc import ABC, abstractmethod
+
+
+class BaseManager(ABC):
+    """The base class for managers."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the BaseManager."""
+        super(BaseManager, self).__init__(*args, **kwargs)
+
+    @abstractmethod
+    def add(self):
+        """Abstract method to add a computation to the manager."""
+        pass
```

Comment:
Do you want to `raise UnimplementedError' for these abstract methods? I think sometimes abstract classes do that to make it obvious to callers when interfaces were not properly implemented by the child class

Reply:
seems it is discouraged:
https://stackoverflow.com/questions/45826692/body-of-abstract-method-in-python-3-5
https://stackoverflow.com/questions/60726156/best-way-to-implement-abstract-classes-in-python
actually I will just remove the pass as recommended in the first link, it seems it is redundant if you have doc strings

Reply:
Oh! Good to know, thanks for sending the links